parquet-avro/src/test/java/org/apache/parquet/avro/TestArrayCompatibility.java

```diff
@@ -56,6 +59,30 @@ public static void setupNewBehaviorConfiguration() {
         AvroSchemaConverter.ADD_LIST_ELEMENT_RECORDS, false);
   }
 
+  @Test
+  public void testReadEmptyParquetFileWriteNull() throws IOException {
+    final Schema schema;
+    try (InputStream avroSchema = Resources.getResource("persons.json").openStream()) {
+      schema = new Schema.Parser().parse(avroSchema);
+    }
+
+    try (ParquetWriter<GenericRecord> writer =
+           AvroParquetWriter.<GenericRecord>builder(new org.apache.hadoop.fs.Path("/tmp/persons.parquet"))
+             .withSchema(schema)
+             .build()) {
+
+      // To trigger exception, add array with null element.
+      try {
+        writer.write(new GenericRecordBuilder(schema).set("address", Arrays.asList("first", null, "last")).build());
+      } catch (NullPointerException e) {
```

Comment:
I would add a `fail` in the `try` block after write so we fail if no exception occurred.